packages/core/src/pages/api/graphql.ts

```diff
@@ -59,6 +59,12 @@ const handler: NextApiHandler = async (request, response) => {
         ? stringifyCacheControl(extensions.cacheControl)
         : 'no-cache, no-store'
 
+    if (extensions.cookies && !hasErrors) {
+      extensions.cookies.forEach((cookie) =>
+        response.setHeader('set-cookie', cookie)
+      )
```

Comment:
The existing code will override the `setCookie` in every loop iteraction. We need to pass the string[] with the cookies directly as the second argument.
```suggestion
      response.setHeader('set-cookie', extensions.cookies)
```
Checkout example:
|Before|After|
|-|-|
|<img width="762" alt="Screenshot 2023-12-06 at 12 59 18" src="https://github.com/vtex/faststore/assets/11325562/c5c8b2c6-3875-48b4-b6bb-3fc7b64b0de0">|<img width="767" alt="Screenshot 2023-12-06 at 12 58 13" src="https://github.com/vtex/faststore/assets/11325562/d5aadc3c-c797-455d-9ba4-aefc7665d950">|